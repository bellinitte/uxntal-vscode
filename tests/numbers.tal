0 01 012 0123 01234 ( Length of a raw number must be 2 or 4 )
ab aB Ab AB abcd abcD ABCD ( All digits of raw numbers must be lowercase )

#01 #abcd
|01 |abcd
$01 $abcd

( Runed numbers can be of any length )
#0 #012 #01234
|0 |012 |01234
$0 $012 $01234

# | $

( Literal numbers can have arbitrary "digits" )
#0x #x!@ ##

( Paddings cannot )
|0x0
$0x0
