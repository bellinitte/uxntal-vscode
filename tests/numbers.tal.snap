>0 01 012 0123 01234 ( Length of a raw number must be 2 or 4 )
#^ source.tal entity.name.function.uxntal
# ^ source.tal
#  ^^ source.tal constant.numeric.uxntal
#    ^ source.tal
#     ^^^ source.tal entity.name.function.uxntal
#        ^ source.tal
#         ^^^^ source.tal constant.numeric.uxntal
#             ^ source.tal
#              ^^^^^ source.tal entity.name.function.uxntal
#                   ^ source.tal
#                    ^ source.tal comment.uxntal
#                     ^ source.tal
#                      ^^^^^^ source.tal comment.uxntal
#                            ^ source.tal
#                             ^^ source.tal comment.uxntal
#                               ^ source.tal
#                                ^ source.tal comment.uxntal
#                                 ^ source.tal
#                                  ^^^ source.tal comment.uxntal
#                                     ^ source.tal
#                                      ^^^^^^ source.tal comment.uxntal
#                                            ^ source.tal
#                                             ^^^^ source.tal comment.uxntal
#                                                 ^ source.tal
#                                                  ^^ source.tal comment.uxntal
#                                                    ^ source.tal
#                                                     ^ source.tal comment.uxntal
#                                                      ^ source.tal
#                                                       ^^ source.tal comment.uxntal
#                                                         ^ source.tal
#                                                          ^ source.tal comment.uxntal
#                                                           ^ source.tal
#                                                            ^ source.tal comment.uxntal
>ab aB Ab AB abcd abcD ABCD ( All digits of raw numbers must be lowercase )
#^^ source.tal constant.numeric.uxntal
#  ^ source.tal
#   ^^ source.tal entity.name.function.uxntal
#     ^ source.tal
#      ^^ source.tal entity.name.function.uxntal
#        ^ source.tal
#         ^^ source.tal entity.name.function.uxntal
#           ^ source.tal
#            ^^^^ source.tal constant.numeric.uxntal
#                ^ source.tal
#                 ^^^^ source.tal entity.name.function.uxntal
#                     ^ source.tal
#                      ^^^^ source.tal entity.name.function.uxntal
#                          ^ source.tal
#                           ^ source.tal comment.uxntal
#                            ^ source.tal
#                             ^^^ source.tal comment.uxntal
#                                ^ source.tal
#                                 ^^^^^^ source.tal comment.uxntal
#                                       ^ source.tal
#                                        ^^ source.tal comment.uxntal
#                                          ^ source.tal
#                                           ^^^ source.tal comment.uxntal
#                                              ^ source.tal
#                                               ^^^^^^^ source.tal comment.uxntal
#                                                      ^ source.tal
#                                                       ^^^^ source.tal comment.uxntal
#                                                           ^ source.tal
#                                                            ^^ source.tal comment.uxntal
#                                                              ^ source.tal
#                                                               ^^^^^^^^^ source.tal comment.uxntal
#                                                                        ^ source.tal
#                                                                         ^ source.tal comment.uxntal
>
>#01 #abcd
#^ source.tal punctuation.rune.uxntal
# ^^ source.tal constant.numeric.uxntal
#   ^ source.tal
#    ^ source.tal punctuation.rune.uxntal
#     ^^^^ source.tal constant.numeric.uxntal
>|01 |abcd
#^ source.tal punctuation.rune.uxntal
# ^^ source.tal constant.numeric.uxntal
#   ^ source.tal
#    ^ source.tal punctuation.rune.uxntal
#     ^^^^ source.tal constant.numeric.uxntal
>$01 $abcd
#^ source.tal punctuation.rune.uxntal
# ^^ source.tal constant.numeric.uxntal
#   ^ source.tal
#    ^ source.tal punctuation.rune.uxntal
#     ^^^^ source.tal constant.numeric.uxntal
>
>( Runed numbers can be of any length )
#^ source.tal comment.uxntal
# ^ source.tal
#  ^^^^^ source.tal comment.uxntal
#       ^ source.tal
#        ^^^^^^^ source.tal comment.uxntal
#               ^ source.tal
#                ^^^ source.tal comment.uxntal
#                   ^ source.tal
#                    ^^ source.tal comment.uxntal
#                      ^ source.tal
#                       ^^ source.tal comment.uxntal
#                         ^ source.tal
#                          ^^^ source.tal comment.uxntal
#                             ^ source.tal
#                              ^^^^^^ source.tal comment.uxntal
#                                    ^ source.tal
#                                     ^ source.tal comment.uxntal
>#0 #012 #01234
#^ source.tal punctuation.rune.uxntal
# ^ source.tal constant.numeric.uxntal
#  ^ source.tal
#   ^ source.tal punctuation.rune.uxntal
#    ^^^ source.tal constant.numeric.uxntal
#       ^ source.tal
#        ^ source.tal punctuation.rune.uxntal
#         ^^^^^ source.tal constant.numeric.uxntal
>|0 |012 |01234
#^ source.tal punctuation.rune.uxntal
# ^ source.tal constant.numeric.uxntal
#  ^ source.tal
#   ^ source.tal punctuation.rune.uxntal
#    ^^^ source.tal constant.numeric.uxntal
#       ^ source.tal
#        ^ source.tal punctuation.rune.uxntal
#         ^^^^^ source.tal constant.numeric.uxntal
>$0 $012 $01234
#^ source.tal punctuation.rune.uxntal
# ^ source.tal constant.numeric.uxntal
#  ^ source.tal
#   ^ source.tal punctuation.rune.uxntal
#    ^^^ source.tal constant.numeric.uxntal
#       ^ source.tal
#        ^ source.tal punctuation.rune.uxntal
#         ^^^^^ source.tal constant.numeric.uxntal
>
># | $
#^ source.tal punctuation.rune.uxntal
# ^ source.tal
#  ^ source.tal punctuation.rune.uxntal
#   ^ source.tal
#    ^ source.tal punctuation.rune.uxntal
>
>( Literal numbers can have arbitrary "digits" )
#^ source.tal comment.uxntal
# ^ source.tal
#  ^^^^^^^ source.tal comment.uxntal
#         ^ source.tal
#          ^^^^^^^ source.tal comment.uxntal
#                 ^ source.tal
#                  ^^^ source.tal comment.uxntal
#                     ^ source.tal
#                      ^^^^ source.tal comment.uxntal
#                          ^ source.tal
#                           ^^^^^^^^^ source.tal comment.uxntal
#                                    ^ source.tal
#                                     ^^^^^^^^ source.tal comment.uxntal
#                                             ^ source.tal
#                                              ^ source.tal comment.uxntal
>#0x #x!@ ##
#^ source.tal punctuation.rune.uxntal
# ^^ source.tal constant.numeric.uxntal
#   ^ source.tal
#    ^ source.tal punctuation.rune.uxntal
#     ^^^ source.tal constant.numeric.uxntal
#        ^ source.tal
#         ^ source.tal punctuation.rune.uxntal
#          ^ source.tal constant.numeric.uxntal
>
>( Paddings cannot )
#^ source.tal comment.uxntal
# ^ source.tal
#  ^^^^^^^^ source.tal comment.uxntal
#          ^ source.tal
#           ^^^^^^ source.tal comment.uxntal
#                 ^ source.tal
#                  ^ source.tal comment.uxntal
>|0x0
#^ source.tal punctuation.rune.uxntal
# ^^^ source.tal entity.name.function.uxntal
>$0x0
#^ source.tal punctuation.rune.uxntal
# ^^^ source.tal entity.name.function.uxntal
>